src/librustc/ty/mod.rs

```diff
@@ -578,6 +590,22 @@ impl <'gcx: 'tcx, 'tcx> Canonicalize<'gcx, 'tcx> for Ty<'tcx> {
 #[derive(Debug, RustcEncodable)]
 pub struct Slice<T>([T]);
 
+impl<T> Ord for Slice<T> where T: Ord {
+    fn cmp(&self, other: &Slice<T>) -> Ordering {
+        if self == other { Ordering::Equal } else {
+            <[T] as Ord>::cmp(&self.0, &other.0)
+        }
+    }
+}
+
+impl<T> PartialOrd for Slice<T> where T: PartialOrd {
+    fn partial_cmp(&self, other: &Slice<T>) -> Option<Ordering> {
+        if self == other { Some(Ordering::Equal) } else {
+            <[T] as PartialOrd>::partial_cmp(&self.0, &other.0)
+        }
```

Comment:
Same here (`Some(self.cmp(other))`).

Reply:
`Some(self.cmp(other))` will require the bound `T: Ord`, which I was not sure we wanted to impose for all types that that maybe only want to implement `PartialOrd`?

Reply:
Ah, I somehow missed that this was generic - seems fine, then.